lib/blueprinter/configuration.rb

```diff
@@ -1,16 +1,28 @@
 module Blueprinter
   class Configuration
-    attr_accessor :generator, :method, :sort_fields_by
+    attr_accessor :generator, :if, :method, :sort_fields_by, :unless
+
+    VALID_CALLABLES = %i(if unless)
 
     def initialize
       @generator = JSON
+      @if = nil
       @method = :generate
       @sort_fields_by = :name_asc
+      @unless = nil
     end
 
     def jsonify(blob)
       generator.public_send(method, blob)
     end
+
+    def valid_callable?(callable_name)
+      VALID_CALLABLES.include?(callable_name)
+    end
+  end
+
+  def self.reset_configuration!
+    @configuration = Configuration.new
```

Comment:
What's the use for this? Is it only to clear the config for the tests?

Reply:
I'm glad you pointed this out -- meant to leave a comment. That was the main primary motivating factor was that I wanted a more consistent way to reset back to the defaults. I re-wrote this to use a SpecHelper that does not modify the original class. Thanks for pointing out this code smell 👃